infrastructure-playbooks/rolling_update.yml

```diff
@@ -401,46 +401,35 @@
 
   pre_tasks:
     - name: get fsmap
-      command: ceph --cluster {{ cluster }} fs dump --format=json
-      delegate_to: "{{ groups[mon_group_name][0] }}"
-      register: get_fsmap
-      when: not containerized_deployment
-
-    - name: get fsmap (containerized)
-      command: docker exec ceph-mon-{{ hostvars[mon_host]['ansible_hostname'] }} ceph --cluster {{ cluster }} fs dump --format=json
-      delegate_to: "{{ groups[mon_group_name][0] }}"
-      register: get_fsmap
-      when: containerized_deployment
+      include_tasks: fs_status.yml
 
-    - name: set_fact fsmap
+    - name: set_fact pre_upgrade_fsmap
       set_fact:
-        fsmap: "{{ get_fsmap.stdout | from_json }}"
+        pre_upgrade_fsmap: "{{ fsmap }}"
 
     - name: reduce ranks to 1
-      command: ceph --cluster {{ cluster }} fs set {{ fsmap['mds_map']['fs_name'] }} max_mds 1
+      command: ceph --cluster {{ cluster }} fs set {{ pre_upgrade_fsmap['mds_map']['fs_name'] }} max_mds 1
       delegate_to: "{{ groups[mon_group_name][0] }}"
       when: not containerized_deployment
 
     - name: reduce ranks to 1 (containerized)
-      command: docker exec ceph-mon-{{ hostvars[mon_host]['ansible_hostname'] }} ceph --cluster {{ cluster }} fs set {{ fsmap['mds_map']['fs_name'] }} max_mds 1
+      command: docker exec ceph-mon-{{ hostvars[mon_host]['ansible_hostname'] }} ceph --cluster {{ cluster }} fs set {{ pre_upgrade_fsmap['mds_map']['fs_name'] }} max_mds 1
       delegate_to: "{{ groups[mon_group_name][0] }}"
       when: containerized_deployment
 
-    - name: wait for stable fsmap
-      command: ceph --cluster {{ cluster }} fs dump --format=json
-      delay: 5
-      delegate_to: "{{ groups[mon_group_name][0] }}"
-      register: get_fsmap
-      until: {{ get_fsmap.stdout | from_json }}['mds_map']['in'] == [0]
-      when: not containerized_deployment
+    - name: deactivate all non-zero ranks
```

Comment:
@leseb, I need some ansible expertise for this PR. This loop needs to try to repeatedly deactivate the highest rank and then wait for the FSMap to stabilize (i.e. the highest non-zero rank fully deactivates). This requires running a sequence of tasks (mds_pre_upgrade.yml) until a condition is met: `highest_rank == 0`. Unfortunately, it doesn't seem possible to run a sequence of tasks in a loop: there appears to be a bug where `include_tasks` works as expected but the `when:` clause is ignored so the "loop" always runs once.

How should I go about fixing this?

Reply:
@batrick according to this issue https://github.com/ansible/ansible/issues/17098 it seems this is still not supported...
At the moment, I don't see an easier workaround than what is proposed by @nerdvegas in the issue

Reply:
Do we have a suggessted approach for rurnning a bash script on the mons? e.g. where do we store the script? Some existing code to look at?

@guits @leseb

Reply:
You can write a jinja template similar to this https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-defaults/templates/restart_mon_daemon.sh.j2 and then run it like this: https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-defaults/handlers/main.yml#L28

Does that help?